lighthouse-core/lib/icons.js

```diff
@@ -33,13 +33,13 @@ function pngSizedAtLeast(sizeRequirement, manifest) {
   /** @type {Array<string>} */
   const flattenedSizes = [];
   iconValues
+    // filter out icons with no typehint + extension is not .png
     // filter out icons with a typehint that is not 'image/png'
-    .filter(icon => (!icon.value.type.value) ||
+    .filter(icon => (!icon.value.type.value && (icon.value.src.value &&
```

Comment:
this is long enough now that we should just make a function block. It's more verbose, but what about something like
```js
.filter(icon => {
  const typeHint = icon.value.type.value;
  if (typeHint) {
    // if a type hint is present, filter out icons that are not 'image/png'.
    return typeHint === 'image/png';
  }

  // otherwise, fall back to filtering on the icons' extension.
  const src = icon.value.src.value;
  return src && new URL(src).pathname.endsWith('.png');
}).forEach(icon => {
  // ...
```